### ISSUE TITLE
onStateChange never called for <NavigationContainer independent/>

### ISSUE THREAD

**Issue by retyui** [react-navigation/react-navigation#8354]
**Current Behavior**

I tried to create a Slit view  with a separate NavigationContainer
And I found out that is buggy because `onStateChange`  always called with `undefined` when I push some screens

```js
// Example of virtual dom tree
// All code you will see in "How to reproduce" section!
<NavigationContainer>
  <Tab.Navigator>
    <Tab.Screen>
      <LiftStack.Navigator />
      <NavigationContainer
        independent
        onStateChange={(state) => {
          // state always `undefined`
          console.log(' --- onStateChange: IndepNav', state);
        }}
      >
        <RightStack.Navigator />
      </NavigationContainer>
    </Tab.Screen>
  </Tab.Navigator>
</NavigationContainer>
```

**Expected Behavior**

onStateChange should work properly 

**How to reproduce**

1) Please use the code below.
2) The run application in debug mode then open `Home` tab
3) Then press buttons "Push ... to LEFT\RIGHT" 
4) Check console you will see that `onStateChange` for an independent NavigationContainer always `undefined` 

![image](https://user-images.githubusercontent.com/4661784/83172080-b753c500-a11f-11ea-9cec-4273ee019d43.png)

<details>
  <summary>!!! CODE HERE !!!</summary>
  

  
```js
  import 'react-native-gesture-handler';
  import { AppRegistry, Text, View } from 'react-native';

  import React from 'react';

  import { NavigationContainer, StackActions } from '@react-navigation/native';
  import { createStackNavigator } from '@react-navigation/stack';

  import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';

  const Tab = createBottomTabNavigator();
  const RightStack = createStackNavigator();
  const LiftStack = createStackNavigator();

  function Content() {
    return (
      <>
        <Text
          onPress={() => {
            independentNavigator.dispatch(StackActions.push('R: Details'));
          }}
        >
          Push "Details" screen to RIGHT Stack
        </Text>
        <Text
          onPress={() => {
            mainNavigator.dispatch(StackActions.push('L: Details'));
          }}
        >
          Push "Details" screen to LEFT Stack
        </Text>
      </>
    );
  }

  function RightStackScreen() {
    return (
      <RightStack.Navigator>
        <RightStack.Screen name="R: Right" component={Content} />
        <RightStack.Screen name="R: Details" component={Content} />
      </RightStack.Navigator>
    );
  }
  function LeftStackScreen() {
    return (
      <LiftStack.Navigator>
        <LiftStack.Screen name="L: Left" component={Content} />
        <LiftStack.Screen name="L: Details" component={Content} />
      </LiftStack.Navigator>
    );
  }

  function Home() {
    return (
      <View style={{ flex: 1, flexDirection: 'row' }}>
        <View style={{ flex: 1, borderWidth: 1, borderColor: 'red' }}>
          <LeftStackScreen />
        </View>
        <View style={{ flex: 1, borderWidth: 1, borderColor: 'blue' }}>
          <NavigationContainer
            independent
            onStateChange={(state) => {
              console.log(' --- onStateChange: IndepNav', state);
            }}
            ref={(ref) => {
              global.independentNavigator = ref;
            }}
          >
            <RightStackScreen />
          </NavigationContainer>
        </View>
      </View>
    );
  }

  function App() {
    return (
      <NavigationContainer
        onStateChange={(ss) => {
          global.sss = ss;
          console.log(' --- onStateChange MainNav', ss);
        }}
        ref={(ref) => {
          global.mainNavigator = ref;
        }}
      >
        <Tab.Navigator>
          <Tab.Screen name="Settings" component={() => <Text>Settings</Text>} />
          <Tab.Screen name="Home" component={Home} />
        </Tab.Navigator>
      </NavigationContainer>
    );
  }

  AppRegistry.registerComponent('App', () => App);
  ```
  
</details>

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 |  Android (maybe iOS) |
| @react-navigation/native       |    5.4.0 |
| react-native                   | 0.60.x |
| expo                           |
| node                           |
| npm or yarn                    | yarn |

**Comment by sienkiewiczkm** [comment]
@retyui Have you found a workaround for this? I think I am experiencing this on `5.8.6`.

**Comment by amorino** [comment]
I have the same problem.

**Comment by hellogerard** [comment]
I believe this was fixed as part of #9080. Looking at the [CHANGELOG](https://github.com/react-navigation/react-navigation/blob/main/packages/core/CHANGELOG.md), it will be released in 6.0.0, but looks like you can try it out if you check out `6.0.0-next.0`. Can anyone verify this?
